test/food-router-test.js

```diff
@@ -0,0 +1,140 @@
+
+require('../server.js');
+
+const url = `http://localhost:${PORT}`;
+const exampleFood = {
+  name: 'test food name',
+  meal: 'Breakfast'
+};
+
+describe('Food Routes', function() {
+  describe('POST: /api/food', function() {
+    describe('with a valid body', function(){
+      after( done => {
+        if (this.tempFood) {
+          Food.remove({})
+          .then( () => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should return a food', done => {
+        request.post(`${url}/api/food`)
+        .send(exampleFood)
+        .end((err, res) => {
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('test food name');
+          expect(res.body.meal).to.equal('Breakfast');
+          this.tempFood = res.body;
+          done();
+        });
+      });
+    });
+
+    describe('with an invalid body', function() {
+      it('should return 400', done => {
+        request.post(`${url}/api/food`)
+        .send({})
+        // ???
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+  describe('GET: /api/food/:id', function() {
+    describe('with a valid body', function() {
+      before( done  => {
+        exampleFood.timestamp = new Date();
+        new Food(exampleFood).save()
+        .then( food => {
+          this.tempFood = food;
+          done();
+        })
+        .catch(done);
+      });
+
+      after(done => {
+        delete exampleFood.timestamp;
+        if(this.tempFood) {
+          Food.remove({})
+          .then( () => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+
+      it('should return a food', done => {
+        request.get(`${url}/api/food/${this.tempFood._id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('test food name');
+          expect(res.body.meal).to.equal('Breakfast');
+          done();
+        });
+      });
+    });
+    describe('with an invalid request', function() {
+      it('should respond with 404', done => {
+        request.get(`${url}/api/fo`)
+        .end(res => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+  });
+  describe('PUT: /api/food/:id', function() {
+    describe('with a valid body', function() {
+      before( done => {
+        exampleFood.timestamp = new Date();
+        new Food(exampleFood).save()
+         .then( food => {
+           this.tempFood = food;
+           done();
+         })
+         .catch(done);
+      });
+      after( done => {
+        if(this.tempFood) {
+          Food.findByIdAndRemove(this.tempFood._id)
+           .then( () => done())
+           .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should respond with 200', done => {
```

Comment:
This is a false positive test. If you console.log the res you'll notice that the res.body has not been updated with the new name and meal.